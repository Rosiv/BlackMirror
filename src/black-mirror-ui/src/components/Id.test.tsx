import * as React from 'react';
import renderer from 'react-test-renderer';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
const customMuiTheme = getMuiTheme(darkBaseTheme);

import Id from './Id';

it('renders and is compatible with the snapshot', () => {
    const tree = renderer
        .create(
        <MuiThemeProvider muiTheme={customMuiTheme}>
            <Id
                value={'012345'}
                nestedLevel={0}
            />
        </MuiThemeProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
