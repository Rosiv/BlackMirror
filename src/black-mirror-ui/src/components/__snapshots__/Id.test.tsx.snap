// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and is compatible with the snapshot 1`] = `
<div>
  <span
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(255, 255, 255, 1)",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "marginLeft": 0,
            "muiPrepared": true,
            "paddingBottom": 16,
            "paddingLeft": 72,
            "paddingRight": 16,
            "paddingTop": 20,
            "position": "relative",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "display": "block",
              "fill": "#ffeb3b",
              "height": 24,
              "left": 4,
              "margin": 12,
              "muiPrepared": true,
              "position": "absolute",
              "top": 12,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
        <div
          style={
            Object {
              "muiPrepared": true,
            }
          }
        >
          012345
        </div>
        <div
          style={
            Object {
              "WebkitBoxOrient": null,
              "WebkitLineClamp": null,
              "color": "rgba(255, 255, 255, 0.7)",
              "display": null,
              "fontSize": 14,
              "height": 16,
              "lineHeight": "16px",
              "margin": 0,
              "marginTop": 4,
              "muiPrepared": true,
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          Id
        </div>
      </div>
    </div>
  </span>
</div>
`;
